src/compiler/parser.ts

```diff
@@ -1822,20 +1824,37 @@ namespace ts {
             }
 
             let tokenPos = scanner.getTokenPos();
+            let nextCharCode: number;
+
             nextToken();
             finishNode(node);
 
-            // Octal literals are not allowed in strict mode or ES5
-            // Note that theoretically the following condition would hold true literals like 009,
-            // which is not octal.But because of how the scanner separates the tokens, we would
-            // never get a token like this. Instead, we would get 00 and 9 as two separate tokens.
-            // We also do not need to check for negatives because any prefix operator would be part of a
-            // parent unary expression.
-            if (node.kind === SyntaxKind.NumericLiteral
-                && sourceText.charCodeAt(tokenPos) === CharacterCodes._0
-                && isOctalDigit(sourceText.charCodeAt(tokenPos + 1))) {
-
-                node.flags |= NodeFlags.OctalLiteral;
+            if (node.kind === SyntaxKind.NumericLiteral) {
+                // Octal literals are not allowed in strict mode or ES5
+                // Note that theoretically the following condition would hold true literals like 009,
+                // which is not octal.But because of how the scanner separates the tokens, we would
+                // never get a token like this. Instead, we would get 00 and 9 as two separate tokens.
+                // We also do not need to check for negatives because any prefix operator would be part of a
+                // parent unary expression.
+                if (sourceText.charCodeAt(tokenPos) === CharacterCodes._0 && isOctalDigit(sourceText.charCodeAt(tokenPos + 1))) {
+                    node.flags |= NodeFlags.OctalLiteral;
+                }
+                else if (token === SyntaxKind.DotToken) {
+                    // Issue #2632 Keep space for  3 .toString()
+                    if (isWhiteSpace(sourceText.charCodeAt(node.end))) {
+						node.end++;
+                        scanner.setTextPos(node.end + 1);
+                    }
+                }
+                else if (token === SyntaxKind.Identifier && sourceText.charCodeAt(node.end-1) === CharacterCodes.dot) {

```

Comment:
i do not really care for this check, `";" expected` or `identifier expected` do not seem too bad in such an obscure error case.